tests/fullydistributed_grids/repartitioning_05.cc

```diff
@@ -0,0 +1,154 @@
+// ---------------------------------------------------------------------
+//
+// Copyright (C) 2021 by the deal.II authors
+//
+// This file is part of the deal.II library.
+//
+// The deal.II library is free software; you can use it, redistribute
+// it, and/or modify it under the terms of the GNU Lesser General
+// Public License as published by the Free Software Foundation; either
+// version 2.1 of the License, or (at your option) any later version.
+// The full text of the license can be found in the file LICENSE.md at
+// the top level directory of deal.II.
+//
+// ---------------------------------------------------------------------
+
+
+// Test
+// TriangulationDescription::Utilities::create_description_from_triangulation()
+// so that it also works for p:d:T set up on a subcommunicator.
+
+#include <deal.II/base/mpi_consensus_algorithms.h>
+
+#include <deal.II/distributed/fully_distributed_tria.h>
+#include <deal.II/distributed/tria.h>
+
+#include <deal.II/fe/fe_q.h>
+
+#include <deal.II/grid/grid_generator.h>
+#include <deal.II/grid/grid_out.h>
+#include <deal.II/grid/grid_tools.h>
+#include <deal.II/grid/tria_description.h>
+
+#include <deal.II/lac/la_parallel_vector.h>
+
+#include <deal.II/numerics/data_out.h>
+
+#include "./tests.h"
+
+using namespace dealii;
+
+MPI_Comm
+create_sub_comm(const MPI_Comm &comm, const unsigned int size)
+{
+  const auto rank = Utilities::MPI::this_mpi_process(comm);
+
+  int color = rank < size;
+
+  MPI_Comm sub_comm;
+  MPI_Comm_split(comm, color, rank, &sub_comm);
+
+  if (rank < size)
+    return sub_comm;
+  else
+    return MPI_COMM_NULL;
```

Comment:
Before I go on with the review, let me ask a question (I also wondered when implementing related functionality): Aren't you throwing away the ranks `>= size`, but without freeing the associated communicator? I don't know all MPI semantics, but I would have expected that I need to call `MPI_Comm_free(&sub_comm)` in the `else` branch. If MPI wants to free all communicators coming out of a `MPI_Comm_split` at the same time, then we would need to return the `sub_comm` part and pick up a different decision.

Reply:
Fixed.